### PR TITLE
Add <code> to sensitive tags in collapse_whitespace_filter.

### LINKED ISSUE #1772: collapse_whitespace in <code>
The documentation says:

> ..is normally ignored by the browser outside of tags like &lt;pre> and <textarea>, one can use CSS properties such as "white-space: pre" to make the browser preserve whitespace within a portion of the document..

It would be nice if collapse_whitespace would also ignore &lt;code> tags, as these are quite a common case for "white-space: pre".

Additionally, MDN says:

> The HTML &lt;code> element displays its contents styled in a fashion intended to indicate that the text is a short fragment of computer code.

Based on that description, it's plausible that a browser may choose to pay attention to whitespace, where it normally wouldn't (given that, for example, Python code without the whitespace would be invalid code).

So, skipping &lt;code> content would make this filter a fair bit safer to use.

### PR DESCRIPTION
**Pull request by Dreamsorcerer**
Fixes #1772

### CODE REVIEW

File: pagespeed/kernel/html/collapse_whitespace_filter.cc [hunk 1]
```diff
@@ -39,7 +39,7 @@ namespace {
 // Tags within which we should never try to collapse whitespace (note that this
 // is not _quite_ the same thing as kLiteralTags in html_lexer.cc):
 const HtmlName::Keyword kSensitiveTags[] = {
-  HtmlName::kPre, HtmlName::kScript, HtmlName::kStyle, HtmlName::kTextarea
+  HtmlName::kCode, HtmlName::kPre, HtmlName::kScript, HtmlName::kStyle, HtmlName::kTextarea
```
Comment (jmarantz) [inline pagespeed/kernel/html/collapse_whitespace_filter.cc:42]: very cool.  Could you also add a test in collapse_whitespace_filter_test.cc?  The pattern should be fairly obvious there.

**Comment by oschaaf** [comment]
Nice, thanks!

**Comment by jmarantz** [comment]
@oschaaf do you know why CI is failing?  I re-started it once, but both attempts ended in:

ld/apache/incubator-pagespeed-mod/third_party/giflib/src'...
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated

I'll try a third time. If that fails we can I guess patch/build/test and then merge, but I wanted to see if you thought the CI was failing like this generally and if we should so something about that.

**Comment by oschaaf** [comment]
@jmarantz I noticed the same with another PR lately, but have not investigated yet. perhaps one of the dependencies broke, I’ll take a look later today

**Comment by oschaaf** [comment]
So the test failed on `But do proxy content if the Content-Type header is present.`.
This was very reproducible because a required script that serves test responses was not running on modpagespeed.com. I recently rebuild the machine from scrap to simplify some things, and forgot all about that script. Restarted the CI job, expect this to be fixed.
(Though sometimes CI may still flake because of the tests requiring an amount of time that is pretty close to the timeout limit.)

**Comment by oschaaf** [comment]
@jmarantz +1 for merging

With regard to giflib, that started failing when sourceforge starting migrating
datacenters. I already attempted a fix [1] earlier because the previous location started 
responding with 404 .. 
But apparently the git repo we now point to times out every now and then, ugh.

[1] https://github.com/apache/incubator-pagespeed-mod/commit/8196fc74847e85085fa6f310d658842261acb329

**Pull request merged**